### ISSUE TITLE
Uncheat player_sprintunlimited

### ISSUE THREAD

**Issue by diamante0018** [iw4x/iw4x-client#61]
**What problem will this solve?**
Marathon perk for everyone without using devmap

**What might be a solution?**
Uncheat player_sprintunlimited

**What other alternatives have you already considered?**
There are none because using devmap allows players to use other unwanted cheats

**Anything else we should know?**
No

**Comment by Rackover** [comment]
After discussions on the discord we advised that the better alternative would be to unprotect SV_CHEATS and allow server owners to change it at runtime just like in source engine, but this would need a bit of testing and/or code as the value might be cached
TODO

**Comment by diamante0018** [comment]
A lot of dvars where unnecessarily cheat protected in playermovement.cpp

In other words, dvars that were completely server-sided were cheat protected.

https://github.com/iw4x/iw4x-client/blob/develop/src/Components/Modules/PlayerMovement.cpp#L306-L342

Only recoil dvars needs to be cheat protected. the code change should be reverted and the other dvars should return to codinfo flag

**Comment by diamante0018** [comment]
server owners can't disable player collision anymore without running devmap which is incredibly hard to work with because enabling sv-cheats allows players to do a bunch of other stuff like using god commands. Server owners need to be able to freely edit the highlighted dvars without enabling sv_cheats

**Comment by diamante0018** [comment]
bg_recoilmultiplayer's dvar name should be change. it is an incedibly misleading dvar name, BG = both games, it runs on the client only, name  should be changed to cg_

**Comment by diamante0018** [comment]
> After discussions on the discord we advised that the better alternative would be to unprotect SV_CHEATS and allow server owners to change it at runtime just like in source engine, but this would need a bit of testing and/or code as the value might be cached TODO

I agree with this, I have now realized this is implemented on iw6-mod and all of Plutonium clients via jump hooks in dvar_variant function. I will try to link that here

**Comment by Rackover** [comment]
![both-is](https://github.com/user-attachments/assets/009d46bd-b1f6-45e1-a826-1343a3dd9bac)

**Comment by diamante0018** [comment]
Requirements.
```
In other games, as far as I know, the host is held to the same "cheat" policy as the clients
If a client or a host wants to edit a cheat dvar, they need sv_cheats to be set to 1, which is in control of the host
The reason sv_cheats is in control of the host is simply because, well, it's a server variable, so the host controls it
The problem so far was that CoD locks-in the sv_cheats variable the moment a server is started, which is annoying for the host
In fact it even copies the "cheating" variable to another memory address, so I'm not sure even setting sv_cheats is enough to "allow cheats" broadly speaking (this is why I thought this would be a tough nut to crack). For some reason i have identified at least two sv_cheats and one dvar_cheat in my game, maybe there's more to replicate (edited)
So the idea would be to:
Allow the host to freely change sv_cheats during a game
Make that change work accross the board (if any other variable needs to be replicated elsewhere then it should be)
Once cheats are enabled on the server, cheat commands/vars become avaialble for both the host and the client
 (edited)
I don't feel like "uncheating every dvar for the host" makes a lot of sense in that context (?)  i'm not even sure why a host would want every dvar uncheated
If you feel like some dvars should not be cheats because hosts tend to change them often, then let us uncheat these (and make them replicated from the server if needed?), as we've done in the past?
```
